hummingbird.py

```diff
@@ -183,20 +183,48 @@ def process_data(self, data):
     if self.start_time is None:
       self.start_time = data.start_time
 
+  def max_of_filtered_arr(self, data, threshold=1):
```

Comment:
This PR is good, but this branch contains old PR#7, you'll need to remove the old PR#7 first.